apps/web/app/query/post.query.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+import { defineQueryOptions } from "./util";
+
+const postsQueryPrefix = "posts";
+
+export const postQueryKey = {
+    id: [postsQueryPrefix, "id"],
+} as const;
+
+export const postQueryOptions = defineQueryOptions({
+    queryKey: postQueryKey.id,
```

Comment:
Why are you using the `postQueryKey.id` as a query key for `getAllPosts`? You're not fetching an individual post by its ID, so it would make more sense to rename this to `postQueryKey.home`, since you're using this query key to fetch posts on the home page:

```suggestion
export const postQueryKey = {
     home: [postsQueryPrefix, "home"],
} as const;

export const postQueryOptions = defineQueryOptions({
    queryKey: postQueryKey.home,
```